asana-helpers-markdown.js

```diff
@@ -119,14 +135,69 @@
     return null;
   }
 
+  const waitForSuccessNotification = () => (new Promise((resolve, _reject) => {
+    const observer = new MutationObserver((mutations) => {
+      const addedNodes = getAddedNodes(mutations);
+
+      const [notification] = queryAll(addedNodes, '.ToastNotificationContent');
+      if (notification) {
+        observer.disconnect();
+        console.debug('=>> Found notification.', { notification });
+        return resolve(notification);
+      }
+      console.debug('=>> Mutation, but notification?', { mutations });
+    });
+
+    const notificationsDiv = document.querySelector('.ToastManager');
+    console.debug('=>> Observing…', notificationsDiv);
+    observer.observe(notificationsDiv, { subtree: true, childList: true });
+  }));
+
+  const getCurrentTaskUrlFromCopyLinkButton = async () => {
+    const observerWaiting = waitForSuccessNotification();
+    document.querySelector('.TaskPaneToolbar-copyLinkButton,.ConversationToolbar-copyLinkButton').click();
+
+    console.info('=>> Waiting for copied link…');
+    await observerWaiting;
+    console.info('=>> Reading copied link…');
+    const href = await navigator.clipboard.readText();
+    console.debug('=>> Read:', { href }); // TODO: should we verify that the href is actually an URL pointing to app.asana.com?
+    return href;
```

Comment:
I initially implemented this more complex mode. One complication I really disliked is that it will only work if the user allows `app.asana.com` to read your clipboard.

Only later, I figure out a better way to get the task id. I'm keeping this worse mode as a fallback, for now. I guess I got some emotional attachment to it, too. 😝

Reply:
This does seem a little complex, but aside from adding extra code the the file I don't see an issue with keeping it for now. I wonder if it is possible to hook into the React DOM/events like the React dev tools do to more easily access some of these properties.